### ISSUE TITLE
e_system function opens cmd on windows

### ISSUE THREAD

**Issue by PetrPilch** [msel-source/meflib#13]
Hi, e_system function is opening cmd on windows and is slowing the opening and buffering data.
We have replaced this function with file_utils_make_directory function from the attached files we created for the similar issues.
[fileutils.zip](https://github.com/msel-source/meflib/files/4543844/fileutils.zip)

**Comment by cimbi** [comment]
Hi, thanks for the comment and the code. I will look into this next week.

**Comment by cimbi** [comment]
Btw I was able to find it only on line 3710. That is the only call of e_system I can find and it is called only if the directory tree does not exist. Can you confirm this is the troublesome piece of code you are dealing with?

**Comment by PetrPilch** [comment]
Yes, this is the place, where we use this file_utils_make_directory function instead of the e_system.

**Comment by cimbi** [comment]
@MaxvandenBoom do you see the same problem on Windows?

**Comment by MaxvandenBoom** [comment]
I have never encountered it with a variety of datasets (on windows). But it seems very specific to a situation where 'A component of the required directory tree does not exist'.

I could check it if @PetrPilch is able to provide a dataset where this occurs?

**Comment by cimbi** [comment]
System calls on windows open cmd window. We've seen this before. I think we should eliminate all system calls if possible. We can fix this in the upcoming PR that will tackle #14 and #15

**Comment by vkremen** [comment]
That would be great guys, to fix all these three things together. Thanks much!

- Vaclav


Vaclav Kremen, M.Sc.Eng., Ph.D.

________________________________________________________________

Associate Professor of Biomedical Engineering, Principal Engineer

@ Mayo Systems Electrophysiology Lab  |  Dept. of Neurology

Phone: (507) 538-5698  |  Pager: 127-09723  |  Email: kremen.vaclav@mayo.edu

Google Scholar: https://orcid.org/0000-0001-9844-7617

________________________________________________________________

Mayo Clinic  |  200 First Street SW, Rochester, MN 55905  |  www.mayoclinic.org


________________________________

**Comment by MaxvandenBoom** [comment]
Jan, I already submitted that PR :)

But I think you are right that these kind of system calls should not happen. I could check and implement something, but that still would require a dataset for me to develop on (@vkremen)

**Comment by vkremen** [comment]
I believe Petr Pilch was submitting the solution. That is the tested one.



- Vaclav
________________________________

**Comment by cimbi** [comment]
I've been playing around with this. The mkdir calls do not match the current code in functionality. The current system call creates the directories recursively. Bud mkdir function would fail in this case.

Perhaps a solution would be something like [this](https://stackoverflow.com/questions/2336242/recursive-mkdir-system-call-on-unix)?

@dancrepeau what is your take on this?

**Comment by dancrepeau** [comment]
I haven't tested the proposed solution (_mkdir), but the documentation seems to be saying that the intermediate directories must already exist:
(from the documentation: "The _mkdir function creates a new directory with the specified dirname. _mkdir can create only one new directory per call, so only the last component of dirname can name a new directory. _mkdir does not translate path delimiters.")

In contrast, the mkdir command (which I think has to be called through the cmd shell) does create the intermediate steps.  Of course, we're trying to remove cmd from being called.

I'm leaning towards something like what Jan suggested, but I'm only basing that on a (somewhat poorly worded) documentation remark about _mkdir.
